lib/trie/child_storage.go

```diff
@@ -16,17 +16,20 @@ var ChildStorageKeyPrefix = []byte(":child_storage:default:")
 var ErrChildTrieDoesNotExist = errors.New("child trie does not exist")
 
 // PutChild inserts a child trie into the main trie at key :child_storage:[keyToChild]
+// A child trie is added as a node (K, V) in main trie. K is the child storage key
+// associated to the child trie, and V is the Merkle value of its corresponding child trie.
+// To get merkle value of a node, we get value of the node. If the value is shorter than
+// 32 bits, we use it as merkle value. Otherwise, we use it's blake2b hash as merkle value.
```

Comment:
- Clarify difference between `node value` and `root hash`, using `value` for both is confusing
- It's 32 bytes, not 32 bits
- Add some details of what is blake2b-ed if the node value is not shorter than 32 bytes
- 

```suggestion
// A child trie is added as a node (K, V) in the main trie. K is the child storage key
// associated to the child trie, and V is the root hash of the child trie.
// If the child trie root node value is shorter than 32 bytes, this one is used as V.
// Otherwise, we use the blake2b hash digest of the child trie root node encoding as V.
```

Reply:
> I think since child tries are tries in themselves, the value (ie. their root hash) should always be a 32-byte hash, right?

True, will change the comment accordingly

Reply:
changed